src/main/java/Duke/Parser.java

```diff
@@ -0,0 +1,134 @@
+package Duke;
+
+/**
+ * Duke.Tokenise is to help split user input
+ * into individual strings for sanitisation
+ * and other checks
+ */
+public class Parser {
+    private String[] tokens = new String[]{};
+    private int markIndex;
+    private String description;
+    private String time;
+
+    public Parser(String userInput) throws DukeException {
```

Comment:
This is constructor is too long ([>30 LoC](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-long-methods)). How about abstracting out code blocks for each case and applying [SLAP](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#slap-hard) here?